tools/java/org/unicode/cldr/tool/ShowLanguages.java

```diff
@@ -1022,22 +1022,9 @@ private String urlEncode(String input) {
             }
         }
 
-        // http://trac.edgewall.org/wiki/TracTickets#PresetValuesforNewTickets
-        // http://unicode.org/cldr/trac/newticket?summary=Fix_XXX
         private String addBug(int bugNumber, String text, String from, String subject, String body) {
-            String parameters = "";
-            // if (from != null && from.length() != 0) {
-            // parameters += "&from=" + urlEncode(from);
-            // }
-            if (body != null && body.length() != 0) {
-                parameters += "&amp;description=" + urlEncode(body);
```

Comment:
It appears from the delta that you are not urlEncoding various fields. Seems dangerous; can you explain?

Reply:
Yes. It looks like formerly, with Trac, the URL could include pre-filled information about the context of the particular bug. Now CLDR_NEWTICKET_URL doesn't accept parameters -- there's no query string. Consequently, the parameters are unused, except for "text", which was never urlEncoded, and doesn't need to be since it's not part of the URL itself. It's safe, since "text" is always supplied by us, not by the user, and always has one of two values: "<i>bug</i>" or "<i>add new</i>".